### DISCUSSION TITLE
Configuring Pythagora with OpenAI API

### DISCUSSION THREAD

**Discussion by tarek101** [Pythagora-io/gpt-pilot#930] [Q&A]
I have simple questions on connecting OpenAI or Anthropic with Pythagora:
For Anthropic:  I entered the Anthropic Key, but I still get an error regarding failure to connect with OpenAI!
For OpenAI: What endpoint should I use for  OpenAI API and how to specify the model to use?

**Comment by leomaxwell973** [comment]
Theres something mixed up in the code on the VS:C edition... at least. I have this same issue, i think BricksLLM might be it trying to call OpenAPI models from Anthropic? I was also able to MiM the keys it's trying to use, and they are both sk-bks-################ types. I am pretty sure anthropic typically has sk-art/sk-anthropic API keys, so this is highly unusual for OpenAI and Anthropic sharing such a similar key (numbers were slightly different, and before anyone asks, no they are not static :P) 
recap for above and the point: anthropic seems to be hardcoded to be treated as if it were openai is what it seems. 

So, with the BricksLLM error im getting always causing a throw of 403, 401, despite even getting my own anthropic key to try and just bypass it, it was still broken. it's not the system, it's not Anthropic, it's just bad code(practices), with far too much caching and variables thrown about, for mundane things like an authorization API call, sheesh.  It was also a horrible mistake to basically make GPT the heavy default, probably what broke anthropic and now, whether it's in the py-code or compiled code.
which then all you can do is re-platform, settle for just GPT and take the absurd loss, or my favorite, abandon ship for something else more put together.

more direct to your questions though on openapi, since that part still works:

Here's some examples from the config's which is where all the model declaration and API keys and basic setup happens (when working)
This is a bare bone to highlight it declares a provider (llm) and you assign the models to agents, by official models + version. which you can look up the versioning of the latest and past models on OpenAI's site I'm sure, anthropic did so good guess they do too.

also, if you need more complicated syntax try a wiki or something or dive the code as there's an "extra" object that has quite a few variables and i believe it has a lot to do with adjusting the API call headers, some neat stuff but poorly executed...

(If you were sold on multiple- choice-models, OpenAPI at least during my tinkering broke maybe... once? and practically fixed itself as I tinkered on, def biased lol.) 

...and not documented thoroughly enough.
```
#config.json#
{
  "llm": {                                   #llm/llm-provider
    "openai": {                              #name one of the 4 hard corded (working) providers
      "base_url": null,                      #base-url is obv the endpoint url, for openai, since its biased, it will work best on null since its default
      "api_key": null,                       #API-Key for Pythag pro, leave null, for users with their own API, plug it in.
      "connect_timeout": 60.0,               #perhaps play with timeouts if its spinning wheels burning tokens or disconnecting too soon
      "read_timeout": 10.0                   # same as above.
    }
  },
  "agent": {                                 # agent, the list and declaring providers are done and focusing on the roster now
    "CodeMonkey": {                          # agent "name", you're supposed to use their "python" name, but it's so jumbled the name-to-files...                             
      "provider": "openai",                  #-- aside probably class objects? but why they are so mismatched from the core. prompt. * Files idk
      "model": "gpt-4-0125-preview",         # Models the important bit, make sure to get the latest/desired version. why these are-not updated---
      "temperature": 0.0                     #---to the newest by default is probably the token anthropic/openai gave them was ver specific?
    },                                       # temp: adjusts bots personality, higher=random/nuts, lower=robotic/facts
    "CodeMonkey.describe_files": {
      "provider": "openai",
      "model": "gpt-3.5-turbo",
      "temperature": 0.0
    }
  }
}
```
